### ISSUE TITLE
Keep the user-facing UI simple

### ISSUE THREAD

**Issue by Madis0** [packwiz/packwiz-vanilla-installer#2]
_Reposting a previously suggested idea to keep track of it:_
The user-facing UI should be simple, similar to Fabric, Forge, Optifine installers.
That could potentially be done with URL parameters, e.g. `https://packwiz-site.example/?pack=https://example.com/pack.toml`
The installer should also display the name of the modpack.

**Comment by comp500** [comment]
![unknown](https://user-images.githubusercontent.com/1138417/140419643-b244a2cb-391c-42f0-bc88-146622fefabe.png)

The above is roughly how it looks like now; what changes do you suggest? For Fabric (not for Forge since I can't easily automate it's install) all of the fields are filled in as soon as you paste the pack URL. Some of the UI elements will change/hide when you select different loader options.

**Comment by Madis0** [comment]
- Pack URL - if possible, take from the download URL somehow; if not - fair enough, keep the textbox for now and maybe there will be some other way to improve its UX later.
- Installation name - take automatically from the pack.toml `name`
- Game directory - set the default in `.minecraft` (and potentially the subfolder you suggested), keep the textbox so it can be overridden
- Mod loader/Minecraft version/Fabric loader version - take automatically from pack.toml, don't show any options in UI
- Launcher path - inherit from game directory, don't ask separately
- Created version name -  take automatically from the pack.toml `name`

Edit: basically as you said "all the fields are filled", so there is no need to show them to the user at all, the user only needs the pack url and game directory fields.

**Comment by comp500** [comment]
Hmm.... I do at least want to show this information, so the user knows what they're installing (and can change it if they need to). Installation name especially so, as someone might want to install the same modpack multiple times.

It's only really the advanced options that I'd be okay with putting behind an extra button, something like this?
![image](https://user-images.githubusercontent.com/1138417/140422952-d00b921d-d530-4aee-ada1-621a828dd653.png)

For the pack URL, I think I can append it to the jar/exe with some JS code on the website, so you could send the user a link like `https://packwiz.infra.link/download/vanilla#https://raw.githubusercontent.com/Madis0/fabulously-optimized/main/Packwiz/1.17.1/pack.toml` to get a download page with a file that autofills the pack URL. 
... or actually on Windows downloaded files will get a Mark of the Web alternate data stream that lets me get the download URL directly, would be better for smartscreen/etc. (but not guaranteed to be supported by all browsers, and only for Windows)

**Comment by Madis0** [comment]
This is how I would imagine it (assuming there is no way to autofill the pack URL):

Before pasting the URL
![image](https://user-images.githubusercontent.com/8611110/140424198-1d8ebf32-9bf2-4b0a-992e-0f691409914e.png)
After pasting the URL
![image](https://user-images.githubusercontent.com/8611110/140424208-6c345960-a365-4c63-b69e-594057d64274.png)

Installation name is not necessary because it can be confusing in terms of support (user can change the name and forget what they changed it to) and because people who want to change it already know how to do that in the launcher.

Mod loader should at most be _just displayed_, because changing it or its versions can break the modpack completely (e.g. changing from Fabric to Forge, changing to an experimental loader version etc).

**Comment by comp500** [comment]
Yeah that makes sense, though for anything other than Fabric it will need a dropdown to select the version to use (and installer link) since I don't automate Forge installs. I will still have a more options button since I think there are still cases where it's useful to configure the loader/version/launcher path.

**Comment by fuzzyweapon** [comment]
Just a random person who will use this tuning in.  I like the idea of simplifying the UI for people I would have to teach to use this, but what comp500 had first time around would be invaluable in pack development.  It would allow me to install the same pack side by side, but bump a version and test it.  This is huge and opens the door for easily doing a lot of best practices QA/dev wise.  Also, some people may use launchers that use weird layouts and might need to be able to have more flexibility to overcome the lack of direct support for each use case by being able to specify paths.

So I have a strong strong strong flag in the race to not have things /removed/ but hidden under advanced interfaces.
